osu.Game/Audio/HitSampleInfo.cs

```diff
@@ -37,6 +37,16 @@ public class HitSampleInfo : ISampleInfo
         /// </summary>
         public int Volume { get; set; }
 
+        /// <summary>
+        /// Whether the sample was artificially added or if it was a beatmap specification.
```

Comment:
This xmldoc feels like a mental shortcut (what connection does "layering" have with "artificially adding a sample"? I wouldn't understand what it means without the PR description) I'd say it needs to mention explicitly that `IsLayered` is a normal hitsample layered over the beatmap one.

Reply:
That's true, what about: Whether the sample is layered over the beatmap ones or not.

Reply:
I'd go for the very explicit
```suggestion
        /// True if the sample is a <c>hitnormal</c> sample added artificially due to the skin having
        /// the <see cref="osu.Game.Skinning.LegacySkinConfiguration.LegacySetting.LayeredHitSounds"/> option set.
```

But then again adding *that* kind of shows that if possible, `IsLayered` shouldn't really be a flag at all as it is out of place in the general `HitSampleInfo` (it's something that has explicitly to do with legacy skins).

Reply:
In that regard I was thinking about making the `LegacyHitSampleInfo` class public and including the `IsLayered` property in it. Would that work or is it a bad idea? 
Incidentally `IsCustom` could also be included in it.